File: movie_back/app/Models/categories_model.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class categories_model extends Model
{
    use HasFactory;
    protected $table = "categories";
    protected $primaryKey = "id";
    protected $fillable = [
        'id',
        'title',
        'image',
        'status',
    ];
    public function subcategories(){
        return $this->hasMany(subcategories_model::class,'categories_id');
    }
}

File: movie/lib/model/categories_model.dart
import 'package:movie/model/subcategories_model.dart';
class categories_model {
  final int id;
  final String image;
  final String title;
  List<subcategories_model> subcategories;
  categories_model({
    required this.id,
    required this.title,
    required this.image,
    required this.subcategories,
  });
  factory categories_model.fromjson(Map<String,dynamic>json){
    return categories_model(
      id:json['id'],
      image:json['image'],
      title:json['title'],
      subcategories: (json['subcategories'] as List).map((subcategories) => subcategories_model.fromjson(subcategories)).toList(),
    );
  }
}
